src/avm2/native.ts

```diff
@@ -1369,71 +1372,127 @@ module Shumway.AVM2.AS {
   export class ASArgumentError extends ASError { }
 
   export class ASRegExp extends ASObject {
-    public static instanceConstructor: any = XRegExp;
-    public static staticNatives: any [] = [XRegExp];
-    public static instanceNatives: any [] = [XRegExp.prototype];
+    public static instanceConstructor: any = ASRegExp;
+    public static staticNatives: any [] = null;
+    public static instanceNatives: any [] = null;
+
+    public global: boolean;
+    public multiline: boolean;
+    public ignoreCase: boolean;
+    public dotall: boolean;
+    public extended: boolean;
+
+    private _captureNames: string[];
+    private _source: string;
+
+    private _jsRegExp: RegExp;
+
+    constructor(pattern: string = '', flags?: string) {
+        false && super();
+        this.dotall = false;
+        this.extended = false;
+        var captureNames = this._captureNames = [];
+        var p = pattern;
+        if (flags) {
+          var f = '';
+          for (var i = 0; i < flags.length; i++) {
+            var flag = flags[i];
+            if (flag === 's') {
+              // With this flag set, . can match the newline character.
+              p = p.replace(/\./g, '[\\s\\S]');
+              this.dotall = true;
+            } else if (flag === 'x') {
+              // With this flag set, you can type spaces in the regular expression, which are
+              // ignored as part of the pattern.
+              p = p.replace(/\s+/g, '');
+              this.extended = true;
+            } else {
+              f += flag;
+            }
+          }
+        }
+        p = p.replace(/\((?!\?)|\(\?P<([\w$]+)>/g, function (match) {
```

Comment:
Something's wrong with this algorithm. In `test 1` of `test/avm2/pass/reg-0`, all group names are set to `?` instead of the proper group names.